fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/GmailBackedPlatformEmailService.java

```diff
@@ -55,31 +54,42 @@ public void sendToUserAccount(String organisationName, String contactName, Strin
 
     @Override
     public void sendDefinedEmail(EmailDetail emailDetails) {
-        final Email email = new SimpleEmail();
         final SMTPCredentialsData smtpCredentialsData = this.externalServicesReadPlatformService.getSMTPCredentials();
 
         final String authuser = smtpCredentialsData.getUsername();
         final String authpwd = smtpCredentialsData.getPassword();
 
-        // Very Important, Don't use email.setAuthentication()
-        email.setAuthenticator(new DefaultAuthenticator(authuser, authpwd));
-        email.setDebug(false); // true if you want to debug
-        email.setHostName(smtpCredentialsData.getHost());
+        JavaMailSenderImpl mailSender = new JavaMailSenderImpl();
+        mailSender.setHost(smtpCredentialsData.getHost()); // smtp.gmail.com
+        mailSender.setPort(Integer.parseInt(smtpCredentialsData.getPort())); // 587
+
+        // Important: Enable less secure app access for the gmail account used in the following authentication
+
+        mailSender.setUsername(authuser); // use valid gmail address
+        mailSender.setPassword(authpwd); // use password of the above gmail account
+
+        Properties props = mailSender.getJavaMailProperties();
+        props.put("mail.transport.protocol", "smtp");
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.debug", "true");
 
         try {
             if (smtpCredentialsData.isUseTLS()) {
-                // FINERACT-1070: NOT email.setSSLOnConnect(true); email.setSslSmtpPort(smtpCredentialsData.getPort());
-                email.setStartTLSRequired(true);
+                if (smtpCredentialsData.getPort().equals("465")) {
```

Comment:
I don't think hardcoding ports is a good idea...

Reply:
We read the port from smtpCredentials and just check the port number to make sure whether we want to enable ssl or not

Reply:
Hmmm... We should not rely on a port number to decide whether to enable SSL. We should simply have a config parameter to turn it on or off. People can choose to run their mail servers on whatever port they feel like...

Reply:
Yuh. I think we don't need these changes. The previous implementation was ok. But I think the port number needs to get changed as "587". (For the host: smtp.gmail.com)

Reply:
> Hmmm... We should not rely on a port number to decide whether to enable SSL. We should simply have a config parameter to turn it on or off. People can choose to run their mail servers on whatever port they feel like...

@ptuomola  please note that 
**useTLS = YES** is a condition that must be set for port 587 and 465 to successfully send mail (as i test it on linux 16 and 18) if one installs ssmpt server

**UseSTARTTLS=YES** if enabled with port 465 in use - mails will not go  yet it is a must for port 587 for mails to go successfully

Reply:
> I don't think hardcoding ports is a good idea...

@ptuomola  i do agree with you and that is why i created an issue an the jira about this https://issues.apache.org/jira/browse/FINERACT-1070 which when checked @vorburger  agreed 

I was thinking that me and @BLasan  were trying to get things done (by hard cording) and once we are thoroughly done, we shall make sure that variables are picked from the database as per the users settings

Reply:
> We read the port from smtpCredentials and just check the port number to make sure whether we want to enable ssl or not

yes @BLasan  - we are reading if the port is 587 then we set UseSTARTTLS=YES if port 465 then we set it to false and this condition works very well in my view as it avoids un-necessary errors that shall come from the servers

Reply:
If this is the case, maybe you can indicate with a comment why this check was done like this so it's clear for anyone going through the code in the future.